code/src/App.js

```diff
@@ -1,9 +1,32 @@
 import React from 'react'
+import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom'
+import { Header } from 'components/Header'
+import MovieList from './pages/MovieList'
+import MovieDetails from './pages/MovieDetails'
+import MovieUpcoming from './pages/MovieUpcoming'
+import NotFound from 'components/NotFound'
 
 export const App = () => {
+
   return (
-    <div>
-      Find me in src/app.js!
-    </div>
+    <>
+      <BrowserRouter>
+          <Header /> 
+          <Switch> {/* tells browserrouter to pick only one of the routes at a time*/}
+            <Route exact path="/">
+              <MovieList 
+              />
+            </Route>  
+            <Route exact path="/upcoming">
+              <MovieUpcoming 
+              />
+            </Route>
+            <Route path="/movies/:movieId" component={MovieDetails} />
+            <Route path="/404" component={NotFound} />
```

Comment:
different approach for routes, next time try to be consistent with your code and use only one of approaches, consistency is a key :)